command/certificate/verify.go

```diff
@@ -164,6 +178,35 @@ func verifyAction(ctx *cli.Context) error {
 		}
 	}
 
+	if validity {
+
+		remainingValidiy := time.Until(cert.NotAfter)
+		totalValidity := cert.NotAfter.Sub(cert.NotBefore)
+
+		percentUsed := ((totalValidity.Hours() - remainingValidity.Hours()) / totalValidity.Hours()) * 100
+
+		red := "\033[31m"
+		green := "\033[32m"
+		yellow := "\033[33m"
+		reset := "\033[0m"
+		
+		if percentUsed >= 100 {
+			fmt.Println(red, "This certificate has already expired.", reset)
+		} else if percentUsed > 90 {
+			fmt.Printf(red,"Leaf is", int(percentUsed), "% through its lifetime.", reset)
+		} else if percentUsed > 66 && percent < 90 {
+			fmt.Printf(yellow,"Leaf is", int(percentUsed), "% through its lifetime.", reset)
+		} else if percentUsed < 66 && percent > 1 {
+			fmt.Printf(green,"Leaf is", int(percentUsed), "% through its lifetime.", reset)
+		} else if percentUsed < 1{
+			fmt.Printf(green,"Leaf is less than 1% through its lifetime.", reset)
+		} else {
+			return errors.Errorf("failure to determine expiration time for certificate")
+		}
+
+		return nil
+	}
+
```

Comment:
I think we should verify the certificate first `cert.VerifyOpts` below, and then if it's valid we can analyze how much of the lifespan has been used. The benefit of this is that we won't have to check if now < cert.NotBefore or if the cert is already expired. The `VerifyOpts` method already does that for us. So only if the certificate is valid do we print expiry information.